modules/avcodec/encode.c

```diff
@@ -162,6 +163,10 @@ static int init_encoder(struct videnc_state *st)
 	if (!st->codec)
 		return ENOENT;
 
+	/* TODO: Only set sz_dst when swscale is active.
+	   Is there a way to check if swscale is loaded? */
+	(void)conf_get_vidsz(conf_cur(), "video_size", &sz_dst);
```

Comment:
Sounds like a good start to check if the config contains the swscale module, and otherwise issue a warning. The code does work without swscale, the visible effect is that video parts are missing.

E.g. configuring rescaling from 1280x720 to 640x360 would then display only the "1st" quater of the picture in the output stream. So it is not "critical", but the user should be informed that his configuration set will lead to a "loss" of video stream information in the output.